### PR TITLE
Feature: Award Data Archive delta files (including deletions)

### PR DESCRIPTION
**Pull request by michaeldhess**
**High level description:**
Updating the delta file generation script to include deletion records.

**Technical details:**
Continuation of the `download/management/commands/populate_monthly_delta_files.py` script. Parses the FPDS/FABS deletion files in S3 to retrieve deletions since the `generate_since` date and appends them to the end of the delta file. Does not generate empty files.

**Link to JIRA Ticket:**
[DEV-911](https://federal-spending-transparency.atlassian.net/browse/DEV-911)

**Related PR:**
[#1232](https://github.com/fedspendingtransparency/usaspending-api/pull/1232)

**The following are ALL required for the PR to be merged:**
- [ ] Backend review completed
- Matview impact assessment completed (N/A)
- Frontend impact assessment completed (N/A)
- [ ] Data validation completed
- API Performance evaluation completed and present (N/A)

### CODE REVIEW

File: usaspending_api/download/management/commands/populate_monthly_delta_files.py [hunk 1]
```diff
@@ -102,25 +118,111 @@ def create_local_file(self, award_type, source_query, source, agency_code):
...
         os.close(temp_sql_file)
         os.remove(temp_sql_file_path)
         shutil.rmtree(working_dir)
 
         return zipfile_path
 
+    def add_deletion_records(self, working_dir, award_type, agency_code, source, generate_since):
+        """ Retrieve deletion files from S3 and append necessary records to the end of the the file """
+        logger.info('Retrieving deletion records from S3 files and appending to the CSV')
+        file_path = os.path.join(working_dir, '{}_{}_delta.csv'.
```
Comment (dpb-bah) [inline usaspending_api/download/management/commands/populate_monthly_delta_files.py]: How is this different from `source_path`? Could this be passed in?

File: usaspending_api/download/management/commands/populate_monthly_delta_files.py [hunk 2]
```diff
@@ -102,25 +118,111 @@ def create_local_file(self, award_type, source_query, source, agency_code):
...
+                # Split unique identifier into usable columns and add unused columns
+                df = df[AWARD_MAPPINGS[award_type]['unique_iden']].apply(lambda x: pd.Series(x.split('_'))) \
+                    .replace('-none-', '', regex=True).replace('-NONE-', '', regex=True) \
+                    .rename(columns=AWARD_MAPPINGS[award_type]['column_headers'])
+
+                # Only include records within the correct agency, and populated files
+                if len(df.index) == 0:
+                    continue
+                if agency_code != 'all':
+                    df = df[df['awarding_sub_agency_code'].isin(subtier_agencies)]
```
Comment (dpb-bah) [inline usaspending_api/download/management/commands/populate_monthly_delta_files.py]: Clarification: we're using `subtier_code`s to check against toptier agencies as toptier agencies could be duplicates?

Comment (michaeldhess) [reply usaspending_api/download/management/commands/populate_monthly_delta_files.py]: The deletion records do not have any information about top tier agencies, but does have sub tier agency information. So what we do here is make sure that the `awarding_sub_agency_code` maps to the top tier agency.

File: usaspending_api/settings.py [hunk 3]
```diff
@@ -45,6 +45,7 @@
 BULK_DOWNLOAD_AWS_REGION = ""
 MONTHLY_DOWNLOAD_S3_BUCKET_NAME = ""
 BROKER_AGENCY_BUCKET_NAME = ""
+FPDS_BUCKET_NAME = ""
```
[the remark this answers is not in the document]
Comment (michaeldhess) [reply usaspending_api/settings.py]: Yes, @willkjackson looked over it.

**Pull request merged**